jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientIdleTimeoutTest.java

```diff
@@ -120,7 +125,8 @@ public boolean handle(Request request, Response response, Callback callback) thr
         assertEquals(HttpStatus.OK_200, response.getStatus());
 
         // Wait for the server's idle timeout to trigger to give it a chance to clean up its resources.
-        assertTrue(serverIdleTimeoutLatch.await(2 * serverIdleTimeout, TimeUnit.MILLISECONDS));
+        Callback callback = await().atMost(2 * serverIdleTimeout, TimeUnit.MILLISECONDS).until(serverCallbackRef::get, notNullValue());
+        callback.failed(new TimeoutException());
```

Comment:
I'm OK with changing this test to not break.....   but then we need to add a test that explicitly stops the server in all the possible states to check that we do not leak.

If we allow leaks when we stop, then all leak checking is pointless, as we only check leaks after stopping.